test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,154 @@
+                                      }),
+                         Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" + USER_DIR,
+                                          "TempDirDoesNotExist", "io"
+                                      }),
+                         Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" + USER_DIR,
+                                          "TempDirDoesNotExist", "nio"
+                                      }));
+    }
 
-            // error message printed only once
-            testMessageCounter(0, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "io", "nio");
-        }
+    public static Stream<Arguments> counterSource() {
+        // standard test with default setting for java.io.tmpdir
+        return Stream.of(Arguments.of(0,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" + tempDir(),
+                                          "TempDirDoesNotExist", "io", "nio"
+                                      }));
     }
 
-    private static void testMessageExist(int exitValue, String errorMsg, String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldContain(errorMsg)
-                .shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("tempDirSource")
+    public void existingMessage(int exitValue, String errorMsg,
+                                String... options) throws Exception {
+       ProcessTools.executeTestJvm(options).shouldContain(errorMsg)
+           .shouldHaveExitValue(exitValue);
     }
 
-    private static void testMessageNotExist(int exitValue, String errorMsg,String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldNotContain(errorMsg).shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("noTempDirSource")
+    public void nonexistentMessage(int exitValue, String errorMsg,
+                                   String... options) throws Exception {
+        ProcessTools.executeTestJvm(options).shouldNotContain(errorMsg)
+            .shouldHaveExitValue(exitValue);
     }
 
-    private static void testMessageCounter(int exitValue,String... options) throws Exception {
+    @ParameterizedTest
+    @MethodSource("counterSource")
+    public  void messageCounter(int exitValue, String... options)
+        throws Exception {
         OutputAnalyzer originalOutput = ProcessTools.executeTestJvm(options);
         List<String> list = originalOutput.asLines().stream().filter(line
-                -> line.equalsIgnoreCase(ioWarningMsg)).collect(Collectors.toList());
+                -> line.equalsIgnoreCase(WARNING)).toList();
         if (list.size() != 1 || originalOutput.getExitValue() != exitValue)
```

Comment:
(preexisting) the exception message doesn't make much sense in the second case (`originalOutput.getExitValue() != exitValue`)